### ISSUE TITLE
Link boundaries don't work with Safari 11.0.1 when using VoiceOver

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#3373]
Noticed while testing #3343 

On Safari 11.0.1 and only when using VoiceOver, the boundaries don't work at all for me: no highlighting is displayed on both user names and links. No "link modal" appears.

They work on Safari Technology Preview – Release 43 (Safari 11.1, WebKit 12605.1.12). 

I understand this is an edge case and tricky to investigate. However, boundaries works nicely in the Classic Editor so I'm wondering if there's any obvious, easy to spot difference (events or something) in the Gutenberg implementation that prevents them to work correctly.

Screenshots:

![screen shot 2017-11-07 at 16 30 42](https://user-images.githubusercontent.com/1682452/32503806-7908e0d0-c3de-11e7-9355-86b5e2264d67.png)

![screen shot 2017-11-07 at 16 30 54](https://user-images.githubusercontent.com/1682452/32503808-7924cdea-c3de-11e7-9c6d-66dc0735573f.png)

![screen shot 2017-11-07 at 16 31 33](https://user-images.githubusercontent.com/1682452/32503810-79405790-c3de-11e7-81e4-33d2d74e6e4b.png)

**Comment by EphoxJames** [comment]
It still seems to work for me though you have to click twice because the first click moves the box and the second click moves the cursor.

**Comment by afercia** [comment]
Oh yes clicking on it works (at the second click). But moving the caret from surrounding text within a link doesn't.

**Comment by afercia** [comment]
Just noticed one more thing not working well when using Safari 11 + VoiceOver that maybe can help understand what's going on. To reproduce:

Move the caret at the end of a paragraph that ends with a link:

![screen shot 2017-11-08 at 09 19 41](https://user-images.githubusercontent.com/1682452/32538514-89988742-c466-11e7-8c30-e654d1d9e742.png)

Press Cmd + A:
- expected: the paragraph content should be selected
- actual: all the blocks get selected (and focus moves to Open Settings Menu?)

![screen shot 2017-11-08 at 09 20 03](https://user-images.githubusercontent.com/1682452/32538566-b1f5aef4-c466-11e7-84b1-855bc3667fd2.png)

Even in this case, can't reproduce with Safari Technology Preview.

**Comment by jeffpaul** [comment]
_This ticket was mentioned in [Slack](https://make.wordpress.org/chat/) in #core-editor by jeffpaul. [View the logs](https://wordpress.slack.com/archives/C02QB2JS7/p1521736622000655)._

**Comment by afercia** [comment]
A few things have changed:
- user mentions don't use links any longer since https://github.com/WordPress/gutenberg/pull/4444
- more importantly, current Safari stable is 11.1 (which was Safari Technology Preview at the time this issue was created) so now boundaries work when using VoiceOver

I've tested also the boundaries used on bold and italic text, they work correctly.
Closing.
